ctrl/MastermindController.php

```diff
@@ -72,24 +72,24 @@ public function start(Form $form): void
      */
     public function play(Form $form): void
     {
-        $mastermind = $this->mastermindManager->find();
-        $combination = [];
-        for ($i = 0; $i < $mastermind->getSize(); $i++) {
-            $combination[] = $form->getValue('numColor' . $i);
-        }
-        $propositions = $mastermind->getPropositions();
-        $propositions[] = new Combination($combination);
-        $mastermind->setPropositions($propositions);
-        $this->mastermindManager->save($mastermind);
-
-        $compareResults = [];
-        $comparator = new CombiComparator($mastermind->getSolution());
-        foreach ($propositions as $proposition) {
-            $compareResults[] = $comparator->compare($proposition);
+        if ($this->mastermindManager->find() !== null) {
+            $mastermind = $this->mastermindManager->find();
+            $combination = [];
+            for ($i = 0; $i < $mastermind->getSize(); $i++) {
+                $combination[] = $form->getValue('numColor' . $i);
+            }
+            $propositions = $mastermind->addProposition(new Combination($combination));
+            $this->mastermindManager->save($mastermind);
+            $compareResults = [];
+            $comparator = new CombiComparator($mastermind->getSolution());
+            foreach ($propositions as $proposition) {
+                $compareResults[] = $comparator->compare($proposition);
+            }
+            $form = new Form();
+            $this->render(ROOT_DIR . 'view/play.php', compact('mastermind', 'compareResults', 'form'));
         }
-        $colors = new Combination(Mastermind::LEVELS[$mastermind->getLevel()]);
-        $form = new Form();
-        $this->render(ROOT_DIR . 'view/play.php', compact('mastermind', 'compareResults', 'colors', 'form'));
+        ErrorManager::add('La partie n\'a pas encore été créé !');
+        $this->home(new Form());
```

Comment:
- Là t'es obligé de faire un else parce qu'il n'y a pas de return ou de throw.
- Pour la redirection je pensais plutôt à quelque chose du genre `header('index.php')` parce que c'est pas cohérent d'avoir la vue 'home' à l'adresse 'play'
- Là  (`$this->home(new Form());`) par contre c'est pas vraiment cohérent de passer ton formulaire en paramètre dans la mesure ou il ne te sert à rien : tu ne fais que le transmettre à ta vue. Mieux vaut que tu le crées dans le corps de la fonction. Et même ça peut être source d'erreur parce que tu peux lancer `$this->home(null);`

Reply:
Le problème de revenir sur l'index direct alors que play envoie target=play fou la merde.
J'ai essayé de faire un $_GET ='' avant le header(ROOT_DIR . 'index.php') mais cela ne change rien.
A moins que le problème vient du fait que ma façon de tester ne soit pas bonne.
Je mets un un session_destroy() en début de fonction play dans le controller.